src/main/java/digital/slovensko/autogram/server/dto/SignRequestBody.java

```diff
@@ -47,10 +57,56 @@ public SigningParameters getParameters() throws RequestValidationException {
 
         parameters.validate(getDocument().getMimeType());
 
-        return parameters.getSigningParameters(isBase64());
+        SigningParameters signingParameters = parameters.getSigningParameters(isBase64());
+
+        validateXml(signingParameters);
+
+        return signingParameters;
     }
 
     private boolean isBase64() {
         return payloadMimeType.contains("base64");
     }
+
+    private void validateXml(SigningParameters signingParameters) {
+        String xsdSchema = signingParameters.getSchema();
+        String xmlContent;
+
+        boolean isXdc = getDocument().getMimeType().equals(AutogramMimeType.XML_DATACONTAINER);
+        if (isXdc) {
+            XDCTransformer xdcTransformer = XDCTransformer.buildFromSigningParametersAndDocument(signingParameters, getDocument());
+            if (!xdcTransformer.validateXsdDigest())
+                throw new RequestValidationException("XML Datacontainer validation failed", "XSD scheme digest mismatch");
+
+            if (!xdcTransformer.validateXsltDigest())
+                throw new RequestValidationException("XML Datacontainer validation failed", "XSLT transformation digest mismatch");
+
+            xmlContent = xdcTransformer.getContentFromXdc();
+        } else {
+            xmlContent = getDecodedContent();
+        }
+
+        if (!validateXmlContentAgainstXsd(xmlContent, xsdSchema)) {
+            throw new RequestValidationException("XML validation failed", "XML validation against XSD failed");
+        }
+    }
+
+    private String getDecodedContent() {
+        return isBase64() ? new String(Base64.getDecoder().decode(document.getContent())) : document.getContent();
+    }
+
+    private boolean validateXmlContentAgainstXsd(String xmlContent, String xsdSchema) {
+        if (xsdSchema == null) {
+            return true;
+        }
```

Comment:
Rozmýšľam, čo to má robiť, keď user pošle XDC bez schémy - či táto celá validácia má byť optional (keďže podporujeme podpis aj nezobraziteľných súborov) alebo ju zakážeme a povieme, že musí vždy s XDC poslať aj vizualizáciu.

Reply:
Tu sa ale nerieši obsah XDC. Tu sa pýta, či v requeste user poslal aj XSD xchému na overenie.